recipes/gnutls/all/conanfile.py

```diff
@@ -0,0 +1,118 @@
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("nettle/3.8.1")
+        self.requires("gmp/6.2.1")
+        self.requires("libiconv/1.17")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.13")
+        if self.options.with_zstd:
+            self.requires("zstd/1.5.2")
+        if self.options.with_brotli:
+            self.requires("brotli/1.0.9")
+
+    def validate(self):
+        if is_msvc(self):
+            raise ConanInvalidConfiguration(f"{self.ref} cannot be deployed by Visual Studio.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self.source_folder)
+
+    def generate(self):
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+        yes_no = lambda v: "yes" if v else "no"
+        autotoolstc = AutotoolsToolchain(self)
+        autotoolstc.configure_args.extend([
+                          "--disable-tests",
+                          "--disable-doc",
+                          "--disable-manpages",
+                          "--disable-full-test-suite",
+                          "--disable-maintainer-mode",
+                          "--disable-option-checking",
+                          "--without-p11-kit",
+                          "--without-idn",
+                          "--with-included-libtasn1",
+                          "--with-included-unistring",
+                          "--with-libiconv-prefix={}".format(self.dependencies["libiconv"].package_folder),
+                          "--enable-shared={}".format(yes_no(self.options.shared)),
+                          "--enable-static={}".format(yes_no(not self.options.shared)),
+                          "--with-cxx={}".format(yes_no(self.options.enable_cxx)),
+                          "--with-zlib={}".format(yes_no(self.options.with_zlib)),
+                          "--with-brotli={}".format(yes_no(self.options.with_brotli)),
+                          "--with-zstd={}".format(yes_no(self.options.with_zstd)),
+                          ])
+
+        autotoolstc.generate()
+        autodeps = AutotoolsDeps(self)
+        autodeps.generate()
+        pkgdeps = PkgConfigDeps(self)
+        pkgdeps.generate()
+
+    def build(self):
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
+
+    def package(self):
+        self.copy("COPYING", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        autotools = Autotools(self)
+        autotools.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["gnutls", "gnutlsxx"] if self.options.enable_cxx else ["gnutls"]
+        self.cpp_info.set_property("cmake_file_name", "GnuTLS")
```

Comment:
Since there's an official one?

```suggestion
        self.cpp_info.set_property("cmake_find_mode", "both")
        self.cpp_info.set_property("cmake_file_name", "GnuTLS")
```

Reply:
makes sense